diff --git a/tools/CMakeLists.txt b/tools/CMakeLists.txt
index d0eb320..5324d07 100644
--- a/tools/CMakeLists.txt
+++ b/tools/CMakeLists.txt
@@ -75,10 +75,28 @@ ecbuild_add_executable( TARGET     gts_count
                         LIBS       ecc_tools )
 
 # grib to netcdf is optional
+set( eccodes_grib_to_netcdf_extra_libs )
+
+if( HAVE_NETCDF )
+  find_package( HDF5 COMPONENTS C HL REQUIRED )
+  find_package( tinyxml2 CONFIG REQUIRED )
+  find_package( CURL REQUIRED )
+  find_package( libaec CONFIG REQUIRED )
+
+  list( APPEND eccodes_grib_to_netcdf_extra_libs
+        hdf5::hdf5_hl
+        HDF5::HDF5
+        tinyxml2::tinyxml2
+        CURL::libcurl
+        libaec::sz )
+endif()
+
 ecbuild_add_executable( TARGET     grib_to_netcdf
                         SOURCES    grib_to_netcdf.cc
                         CONDITION  HAVE_NETCDF
-                        LIBS       ecc_tools NetCDF::NetCDF_C )
+                        LIBS       ecc_tools
+                                   NetCDF::NetCDF_C
+                                   ${eccodes_grib_to_netcdf_extra_libs} )
 
 
 ecbuild_add_executable( TARGET     grib_list_keys
