diff --git a/tools/CMakeLists.txt b/tools/CMakeLists.txt
index d0eb320..e9682ad 100644
--- a/tools/CMakeLists.txt
+++ b/tools/CMakeLists.txt
@@ -75,10 +75,18 @@ ecbuild_add_executable( TARGET     gts_count
                         LIBS       ecc_tools )
 
 # grib to netcdf is optional
+set( eccodes_grib_to_netcdf_libs )
+if( HAVE_NETCDF )
+  find_package(netCDF CONFIG REQUIRED)
+
+  list( APPEND eccodes_grib_to_netcdf_libs
+        netCDF::netcdf )
+endif()
+
 ecbuild_add_executable( TARGET     grib_to_netcdf
                         SOURCES    grib_to_netcdf.cc
                         CONDITION  HAVE_NETCDF
-                        LIBS       ecc_tools NetCDF::NetCDF_C )
+                        LIBS       ecc_tools NetCDF::NetCDF_C ${eccodes_grib_to_netcdf_libs})
 
 
 ecbuild_add_executable( TARGET     grib_list_keys
